include/appbase/application.hpp

```diff
@@ -1,383 +1,23 @@
-          * on first access.  This allows loose and deferred binding between plugins
-          *
-          * @tparam ChannelDecl - @ref appbase::channel_decl
-          * @return reference to the channel described by the declaration
-          */
-         template<typename ChannelDecl>
-         auto get_channel() -> std::enable_if_t<is_channel_decl<ChannelDecl>::value, typename ChannelDecl::channel_type&>
-         {
-            using channel_type = typename ChannelDecl::channel_type;
-            auto key = std::type_index(typeid(ChannelDecl));
-            auto itr = channels.find(key);
-            if(itr != channels.end()) {
-               return *channel_type::get_channel(itr->second);
-            } else {
-               channels.emplace(std::make_pair(key, channel_type::make_unique()));
-               return  *channel_type::get_channel(channels.at(key));
-            }
-         }
-
-         /**
-          * Do not run io_service in any other threads, as application assumes single-threaded execution in exec().
-          * @return io_serivice of application
-          */
-         boost::asio::io_service& get_io_service() { return io_serv; }
-
-         /**
-          * Post func to run on io_service with given priority.
-          *
-          * @param priority can be appbase::priority::* constants or any int, larger ints run first
-          * @param func function to run on io_service
-          * @return result of boost::asio::post
-          */
-         template <typename Func>
-         auto post( int priority, Func&& func ) {
-            return boost::asio::post(io_serv, pri_queue.wrap(priority, std::forward<Func>(func)));
-         }
-
-         /**
-          * Provide access to execution priority queue so it can be used to wrap functions for
-          * prioritized execution.
-          *
-          * Example:
-          *   boost::asio::steady_timer timer( app().get_io_service() );
-          *   timer.async_wait( app().get_priority_queue().wrap(priority::low, [](){ do_something(); }) );
-          */
-         auto& get_priority_queue() {
-            return pri_queue;
-         }
-
-         const bpo::variables_map& get_options() const;
-
-         /**
-          * Set the current thread schedule priority to maximum.
-          * Works for pthreads.
-          */
-         void set_thread_priority_max();
-
-         static void reset_app_singleton() { app_instance.reset(); }
-         static bool null_app_singleton() { return !app_instance; }
-
-      protected:
-         template<typename Impl>
-         friend class plugin;
+// ---------------------------------------------------------------------------------------------
+// to use a different executor:
+//
+//   1. create a copy of this file in your own application
+//
+//   2. create your own executor type (see appbase/default_executor.hpp)
+//      and define an `application` type templated with it.
+//
+//   3. use as below
```

Comment:
How about:
```
// --
// To use a different executor:
//
// 1. Create your own executor type  (see executor types already defined as examples).
// 2. Add a using statement indicating your application and include <appbase/application_instance.hpp>
//          namespace appbase {
//             using application = application_t<default_executor>;
//          }
//          #include <appbase/application_instance.hpp>
// --
```

Reply:
Sure, I have updated the comment, thank you @heifner .